### ISSUE TITLE
bus.end() doesn't work if bus has no subscribers

### ISSUE THREAD

**Issue by AgentME** [baconjs/bacon.js#572]
```
$ node
> var Bacon = require('baconjs');
undefined
> var bus = new Bacon.Bus();
undefined
> bus.ended
false
> bus.end()
undefined
> bus.ended
true
> bus.log('b');
Bacon.Bus()
> bus.push(5);
b 5
[ '<more>' ]
```

I acknowledged this in #517 and wrote it off as a design quirk, but I feel differently now after noticing the misleading value of the `ended` property and having spent a day tracking down an issue that ultimately stemmed from this. (I've worked around it by adding `bus.onValue(function(){});` right after creating the bus.)

**Comment by raimohanska** [comment]
Oh, that's interesting! Would you mind writing a failing test?

**Comment by raimohanska** [comment]
Wow, that was fast :)

**Comment by raimohanska** [comment]
Thanks @phadej for the fix. Released in 0.7.55. Sorry for delay.
